notebook/notebookapp.py

```diff
@@ -43,8 +44,12 @@
 
 # Install the pyzmq ioloop. This has to be done before anything else from
 # tornado is imported.
-from zmq.eventloop import ioloop
-ioloop.install()
+import zmq
+if V(zmq.__version__) < V('17'):
+    from zmq.eventloop import ioloop
+    ioloop.install()
+else:
+    ioloop = None
```

Comment:
I'd be inclined to put the `from tornado` import here directly, so you can see both cases in one snippet.